projects/igniteui-angular/src/lib/chips/chip.spec.ts

```diff
@@ -144,4 +185,167 @@ describe('IgxChip', () => {
+        const firstComponent = components[0];
+
+        expect(firstComponent.componentInstance.displayDensity).toEqual(DisplayDensity.comfortable);
+    });
+
+    it('should make chip comfortable when density is set to comfortable', () => {
+        const fix = TestBed.createComponent(TestChipComponent);
+        fix.detectChanges();
+
+        const components = fix.debugElement.queryAll(By.directive(IgxChipComponent));
+        const secondComponent = components[1];
+
+        expect(secondComponent.componentInstance.displayDensity).toEqual(DisplayDensity.comfortable);
+    });
+
+    it('should make chip compact when density is set to compat', () => {
+        const fix = TestBed.createComponent(TestChipComponent);
+        fix.detectChanges();
+
+        const components = fix.debugElement.queryAll(By.directive(IgxChipComponent));
+        const thirdComponent = components[2];
+
+        expect(thirdComponent.componentInstance.displayDensity).toEqual(DisplayDensity.compact);
+    });
+
+    it('should make chip cosy when density is set to cosy', () => {
+        const fix = TestBed.createComponent(TestChipComponent);
+        fix.detectChanges();
+
+        const components = fix.debugElement.queryAll(By.directive(IgxChipComponent));
+        const fourthComponent = components[3];
+
+        expect(fourthComponent.componentInstance.displayDensity).toEqual(DisplayDensity.cosy);
+    });
+
+    it('should set correctly color of chip when color is set through code', () => {
+        const fix = TestBed.createComponent(TestChipComponent);
+        fix.detectChanges();
+        const chipColor = 'rgb(255, 0, 0)';
+
+        const components = fix.debugElement.queryAll(By.directive(IgxChipComponent));
+        const firstComponent = components[0];
+        const chipAreaElem = firstComponent.queryAll(By.css('.igx-chip__item'))[0];
+
+        firstComponent.componentInstance.color = chipColor;
+
+        expect(chipAreaElem.nativeElement.style.backgroundColor).toEqual(chipColor);
+        expect(firstComponent.componentInstance.color).toEqual(chipColor);
+    });
+
+    it('should delete chip when space button is pressed and chip delete button is focussed', () => {
+        const spaceKeyEvent = new KeyboardEvent('keydown', {
+            'key': ' '
+        });
+
+        const fix = TestBed.createComponent(TestChipComponent);
+        fix.detectChanges();
+
+        let chipComponents = fix.debugElement.queryAll(By.directive(IgxChipComponent));
+
+        expect(chipComponents.length).toEqual(4);
+
+        const deleteButtonElement = fix.debugElement.queryAll(By.css('igx-icon.igx-chip__remove-icon'))[0];
+        deleteButtonElement.nativeElement.focus();
+
+        deleteButtonElement.nativeElement.dispatchEvent(spaceKeyEvent);
+        fix.detectChanges();
+
+        chipComponents = fix.debugElement.queryAll(By.directive(IgxChipComponent));
+        expect(chipComponents.length).toEqual(3);
```

Comment:
@MartinKamenov Check whether the correct chip is deleted. For example filter the model in order to see that there are no objects with id "Country".